src/Common/src/TypeSystem/Ecma/EcmaModule.cs

```diff
@@ -352,11 +353,24 @@ private Object ResolveMemberReference(MemberReferenceHandle handle)
                 }
                 else
                 {
-                    MethodDesc method = ((TypeDesc)parent).GetMethod(name, parser.ParseMethodSignature());
-                    if (method != null)
-                        return method;
+                    MethodSignature sig = parser.ParseMethodSignature();
+                    TypeDesc typeDescToInspect = parentTypeDesc;
 
-                    // TODO: Lookup in parent
+                    // Try to resolve the name and signature in the current type, or any of the base types.
+                    do
+                    {
+                        MethodDesc method = typeDescToInspect.GetMethod(name, sig);
```

Comment:
This will eventually also need to handle substitutions... (just pointing it out - maybe add TODO about it).